overlord/snapstate/handlers.go

```diff
@@ -208,16 +212,25 @@ func linkSnapInFlight(st *state.State, snapName string) (bool, error) {
 			if tc.Kind() == "link-snap" {
 				snapsup, err := TaskSnapSetup(tc)
 				if err != nil {
-					return false, err
+					return nil, err
 				}
 				if snapsup.InstanceName() == snapName {
-					return true, nil
+					tasks = append(tasks, tc)
```

Comment:
because of the conflict checks there shouldn't really be more than one non-ready link-snap task around for a given snap, so this could probably just return at the first hit instead of returning a []*state.Task

Reply:
+1

Reply:
Thanks, fixed this as well in one of the latest commits